### ISSUE TITLE
using  Guns doesnt drop loot

### ISSUE THREAD

**Issue by cool56571** [UchuServer/Uchu#125]

**Comment by Jettford** [comment]
Is this with all ranged weapons or specific ones?

**Comment by cool56571** [comment]
so far ive only tried using it with my flare guns but judging what ive been told its all projectiles

**Comment by TheNexusAvenger** [comment]
I experimented with this a bit. The bug actually is that enemies don't take damage from guns. For some, this is due to a `NullReferenceException` when executing the behavior due to no behavior arguments being populated. I was able to resolve this on my build by adding `castType: SkillCastType.OnUse` in to line 42 in [`Projectile.ImpactAsync`](https://github.com/UchuServer/Uchu/blob/c16eba45265dc84e37cec189182998540b50e7f2/Uchu.World/Objects/Projectile.cs#L42) since the default `OnEquip` wouldn't populate the arguments. However, I have no clue if this is the actual fix because I don't know enough of the system.

Note: I confirmed this with the "Basic Flintlock Pistol" (6937), but the "Elite Long Barrel Blaster" (12729) has a different message (`Invalid behavior sync id: 7!`). The fix addresses some but not all.

Edit: For the `Elite Long Barrel Blaster`, it registers sync ids 2 and 6 but calls 3 for no Imagination and 7 for firing.
